app/src/main/java/com/android/unio/model/user/UserViewModel.kt

```diff
@@ -66,6 +70,21 @@ class UserViewModel(val repository: UserRepository, initializeWithAuthenticatedU
     _user.value?.let { getUserByUid(it.uid, true) }
   }
 
+  fun updateUser(user: User) {
+    repository.updateUser(
+        user,
+        onSuccess = { getUserByUid(user.uid) },
+        onFailure = { Log.e("UserViewModel", "Failed to update user", it) })
+  }
+
+  fun addUser(user: User, onSuccess: () -> Unit) {
+    repository.updateUser(
+        user,
+        onSuccess = onSuccess,
+        onFailure = { Log.e("UserViewModel", "Failed to add user", it) })
+    _user.value = user
+  }
```

Comment:
Minor fix : We could handle the case where the update might fail due to network issues or validation errors, and provide appropriate user feedback through a Toast.